lta/globus_replicator.py

```diff
@@ -141,53 +154,110 @@ async def _quarantine_bundle(self,
             await lta_rc.request('PATCH', f'/Bundles/{bundle["uuid"]}', patch_body)
         except Exception as e:
             self.logger.error(f'Unable to quarantine Bundle {bundle["uuid"]}: {e}.')
+            self.logger.exception(e)
+
+    def _extract_paths(self, bundle: BundleType) -> tuple[Path, Path]:
+        """Get the source and destination paths for the supplied bundle."""
+        source_path = Path(bundle["bundle_path"])
+
+        # destination logic
+        # -- start with basename of /mnt/lfss/jade-lta/bundler_out/fdd3c3865d1011eb97bb6224ddddaab7.zip
+        bundle_name = Path(bundle["bundle_path"]).name
+        if self.use_full_bundle_path:
+            # /data/exp/IceCube/2015/filtered/level2/0320 + BUNDLE_NAME
+            dest_path = Path(bundle["path"]) / bundle_name
+        else:
+            # BUNDLE_NAME
+            dest_path = Path(bundle_name)
+
+        return source_path, dest_path
+
+    async def _patch_bundle(self, lta_rc: RestClient, bundle_id: str, patch_body: dict) -> None:
+        self.logger.info(f"PATCH /Bundles/{bundle_id} - '{patch_body}'")
+        await lta_rc.request('PATCH', f'/Bundles/{bundle_id}', patch_body)
 
     @wtt.spanned()
     async def _replicate_bundle_to_destination_site(self, lta_rc: RestClient, bundle: BundleType) -> None:
         """Replicate the supplied bundle using the configured transfer service."""
         # get our ducks in a row
         bundle_id = bundle["uuid"]
-        bundle_path = bundle["bundle_path"]  # /mnt/lfss/jade-lta/bundler_out/fdd3c3865d1011eb97bb6224ddddaab7.zip
-        # make sure our proxy credentials are all in order
-        self.logger.info('Updating proxy credentials')
-        sgp = SiteGlobusProxy()
-        sgp.update_proxy()
-        # tell GridFTP to 'put' our file to the destination
-        gridftp_dest_url = random.choice(self.gridftp_dest_urls)
-        basename = os.path.basename(bundle_path)
-        if self.use_full_bundle_path:
-            dest_path = bundle["path"]  # /data/exp/IceCube/2015/filtered/level2/0320
-            dest_url = join_smart_url([gridftp_dest_url, dest_path, basename])
-        else:
-            dest_url = join_smart_url([gridftp_dest_url, basename])
-        self.logger.info(f'Sending {bundle_path} to {dest_url}')
+        source_path, dest_path = self._extract_paths(bundle)
+        # -- for mypy, only typehint
+        xfer_identity: TransferIdentityFromGlobus | TransferIdentityFromLTA | None
```

Comment:
Honestly, I'm a little confused on the value-add from the `TransferIdentity*` classes.
I think the logic below could be tightened up a bit. Notes at specific places.